cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx

```diff
@@ -47,15 +47,7 @@ const useAwsDeployment = () => {
             vpc: {
                 cidr: awsData[AWS_FIELDS.VPC_CIDR],
                 vpcID: '',
-                // TODO: single subregion name populated from region selection; but does not support multi-az/HA
-                azs: [
-                    {
-                        name: awsData[AWS_FIELDS.REGION] + 'a',
-                        workerNodeType: nodeType[awsData[AWS_FIELDS.NODE_PROFILE]],
-                        publicSubnetID: '',
-                        privateSubnetID: '',
-                    },
-                ],
+                azs: awsData[AWS_FIELDS.SELECTED_AZS],
```

Comment:
When populate the payload we should be assembling the individual form AZs into the list of objects here. The FORM area should store individual fields for each AZ (AVAILABILITY_ZONE_1, AVAILABILITY_ZONE_2, etc) and should also store individual fields for each AZs related attributes (AVAILABILITY_ZONE_1_WORKER_NODE_INSTANCE_TYPE, etc.)

Then here in the awsDeployment we should assemble the values into their respective objects for each AZ which includes the az name and worker node instance type

Reply:
Got it. Refactored the code